keanu-project/src/test/java/io/improbable/keanu/e2e/regression/RadonHierarchicalRegression.java

```diff
@@ -141,41 +122,44 @@ private RegressionModel createSubModel(double[] allFloor,
         return model;
     }
 
-    private void optimise(BayesianNetwork bayesianNetwork, List<RegressionModel> models) {
+    private void NUTSSample(BayesianNetwork bayesianNetwork) {
+        Vertex muAlpha = bayesianNetwork.getVertexByLabel(new VertexLabel("MuIntercept"));
+        Vertex sigmaAlpha = bayesianNetwork.getVertexByLabel(new VertexLabel("SigmaIntercept"));
+        Vertex muBeta = bayesianNetwork.getVertexByLabel(new VertexLabel("MuGradient"));
+        Vertex sigmaBeta = bayesianNetwork.getVertexByLabel(new VertexLabel("SigmaGradient"));
+
+        NetworkSamples posteriorSamples = NUTS.builder()
+            .maxTreeHeight(5)
+            .saveStatistics(true)
+            .build()
```

Comment:
Does this work with the auto tuning of the step size enabled?

Reply:
seems so, yes, have added